PORTAL/jobs/_pyperformance.py

```diff
@@ -1497,14 +1498,10 @@ def uploadid(self) -> PyperfUploadID:
         try:
             return self._uploadid
         except AttributeError:
-            if self._resfile.uploadid:
-                # XXX Compare with what we get from the metadata?
-                self._uploadid = self._resfile.uploadid
-            else:
-                self._uploadid = PyperfUploadID.from_metadata(
-                    self.metadata,
-                    suite=self.suites,
-                )
+            self._uploadid = PyperfUploadID.from_metadata(
+                self.metadata,
+                suite=self.suites,
+            )
```

Comment:
This was the source of the bug where it wasn't setting that the input results were "multi suite".  There's a note here that says "compare what we get from the metadata", since I think that's more complete/correct.  But by the time we've generated from the metadata, we might as well just always use that -- unless there's another reason you were thinking to compare.

Reply:
> This was the source of the bug where it wasn't setting that the input results were "multi suite".

Yeah, that's what I figured.  My suggestion in https://github.com/faster-cpython/tools/pull/97#discussion_r970066731 should have taken care of it, but your point about verifying is valid.

Logically, if `not self._modified and self._resfile.uploadid` is true then `self._resfile.uploadid` should be the correct value, matching the metadata.  Technically, however, we don't ever verify that a result file's filename matches its content.

So I'm good with dropping the condition as you have and always deriving from the metadata.  It shouldn't be too expensive.  However, it would be worth adding a quick check to make sure the result matches `self._resfile.uploadid` (if applicable).  I'll leave a review comment to demonstrate what I mean.